Automattic/wp-calypso
# SwitcherShell: refactor away from `UNSAFE_componentWillReceiveProps`

#### Changes proposed in this Pull Request

* Migrate `UNSAFE_componentWillReceiveProps` to use `componentDidUpdate` instead

#### Testing instructions

I'm not exactly sure how to test the case we added the `if` clause for. In case a user switches the site the `<Popover />` component is closed anyway which already resets the search state. For now, just make sure those changes make sense. I can't foresee future usages of that component which is why I left the `cdU` call in there instead of removing it.

The `<SwitcherShell />` as part of `<AuthorSelector />` component is used in the following places:

- Delete a user and assign content to another user
- When importing content to a site
- When managing the connection of a Jetpack site

In general the relevant component is only visible when the site has >10 users.

follow up to #51879
